tests/unit/amo/components/TestAddon.js

```diff
@@ -748,6 +771,242 @@ describe('Addon', () => {
+    });
+
+    it('displays the artist name when add-on is a theme', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: fakeTheme,
+        addonsByAuthors: [fakeTheme],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More themes by Madonna');
+    });
+
+    it('displays the author name when add-on is not extension or theme', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeAddon,
+          type: ADDON_TYPE_OPENSEARCH,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More add-ons by Krupa');
+    });
+
+    it('displays a plural form when extension has multiple authors', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More extensions by these authors');
+    });
+
+    it('displays a plural form when theme has multiple authors', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeTheme,
+          authors: Array(2).fill(fakeTheme.authors[0]),
+        },
+        addonsByAuthors: [fakeTheme],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More themes by these authors');
+    });
+
+    it('displays a plural form when add-on has multiple authors', () => {
+      const root = mountMoreAddonsByAuthors({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+          type: ADDON_TYPE_OPENSEARCH,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.Card-header'))
+        .toIncludeText('More add-ons by these authors');
+    });
+
+    it('displays more add-ons by authors', () => {
+      const addonsByAuthors = [
+        { ...fakeAddon, slug: 'addon-1' },
+        { ...fakeAddon, slug: 'addon-2' },
+        { ...fakeAddon, slug: 'addon-3' },
+      ];
+      const root = mountMoreAddonsByAuthors({ addonsByAuthors });
+
+      expect(root.find('.AddonDescription-more-addons--theme'))
+        .toHaveLength(0);
+      expect(root.find(SearchResult)).toHaveLength(addonsByAuthors.length);
+    });
+
+    it('indicates when other add-ons are themes', () => {
+      const addonsByAuthors = [
+        { ...fakeTheme, slug: 'addon-1' },
+        { ...fakeTheme, slug: 'addon-2' },
+        { ...fakeTheme, slug: 'addon-3' },
+      ];
+      const root = mountMoreAddonsByAuthors({
+        addon: fakeTheme,
+        addonsByAuthors,
+      });
+
+      expect(root.find('.AddonDescription-more-addons--theme'))
+        .toHaveLength(1);
+      expect(root.find(SearchResult)).toHaveLength(addonsByAuthors.length);
+    });
+  });
+
+  it('displays a badge when the addon is featured', () => {
```

Comment:
moved the tests below here because they were not in the appropriate `describe()`